src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -85,16 +85,42 @@ internal void Add(HeaderDescriptor descriptor, string? value)
 
             // If we get here, then the value could be parsed correctly. If we created a new HeaderStoreItemInfo, add
             // it to the store if we added at least one value.
-            if (addToStore && (info.ParsedValue != null))
+            if (addToStore && ExistsParsedValue(info))
             {
                 Debug.Assert(!ContainsKey(descriptor));
                 AddEntryToStore(new HeaderEntry(descriptor, info));
             }
         }
 
+        private static bool ExistsParsedValue(HeaderStoreItemInfo info)
```

Comment:
Can these be moved as regular methods on `HeaderStoreItemInfo` instead?

Reply:
The same goes for `ExistsInvalidValue`.

Could these also be renamed to `ContainsParsedValue`/`ContainsInvalidValue`?